### PR TITLE
Reword javadoc in Task for .recover() and .withTimeout()

### PR DESCRIPTION
**Pull request by junchuanwang**
Reworded the javadoc for Task methods so failure handling for timeout task will be clearer.

In short

- Timeout task will be considered failure and will be handled by failure handling methods
- Cancelled task will not be considered as failure for failure handling methods
- Initiating task chain with timeout methods will return a "withtimeout" subtask. When the "withtimeout" subtask timeout, the subtask is not cancelled, but the initiating task is cancelled (this brought some confusion in the doc)

### CODE REVIEW

File: subprojects/parseq/src/main/java/com/linkedin/parseq/Task.java [hunk 1]
```diff
@@ -626,7 +626,7 @@
    * </pre></blockquote>
    * <img src="doc-files/recover-1.png" height="90" width="462"/>
    * <p>
-   * Note that task cancellation is not considered to be a failure.
+   * Note that task cancellation is not considered to be a failure to this method.
```
Comment (karthikbalasub) [inline subprojects/parseq/src/main/java/com/linkedin/parseq/Task.java:629]: I don't think this change is needed. The behavior of cancellation not being considered a failure is not specific to this method.

File: subprojects/parseq/src/main/java/com/linkedin/parseq/Task.java [hunk 2]
```diff
@@ -847,7 +847,7 @@
 
   /**
    * Creates a new task that will handle failure of this task.
-   * Early completion due to cancellation is not considered to be a failure.
+   * Early completion due to cancellation is not considered to be a failure so it will not be recovered.
```
Comment (karthikbalasub) [inline subprojects/parseq/src/main/java/com/linkedin/parseq/Task.java:850]: and the recovery function is not called. (to keep it consistent with the next statement)?

File: subprojects/parseq/src/main/java/com/linkedin/parseq/Task.java (hunk 1 above)
Comment (junchuanwang) [reply subprojects/parseq/src/main/java/com/linkedin/parseq/Task.java:629]: In fact cancelled task is failed, right? It contains EarlyFinishException. I think this confuses people. Also in the user guide it compares cancellation to something like "failure" : https://github.com/linkedin/parseq/wiki/User%27s-Guide#cancelling

I personally think we should treat cancellation as special type of failure that .recover, .recoverWith(), .onFailure() don't care about (just like the doc said).

File: subprojects/parseq/src/main/java/com/linkedin/parseq/Task.java (hunk 2 above)
Comment (junchuanwang) [reply subprojects/parseq/src/main/java/com/linkedin/parseq/Task.java:850]: Will change.

File: subprojects/parseq/src/main/java/com/linkedin/parseq/Task.java (hunk 1 above)
Comment (mchen07) [reply subprojects/parseq/src/main/java/com/linkedin/parseq/Task.java:629]: I see @junchuanwang 's point, it is indeed confusing, especially we have several methods that treats `EarlyFinishException` differently.

**Pull request merged**